### PR TITLE
Updated eventuate-common. Updated to latest version of eventuate-messaging-kafka. Used thread class-loader. https://github.com/eventuate-tram/eventuate…

### PR DESCRIPTION
**Pull request by dartartem**
…-tram-sagas/issues/30

### CODE REVIEW

File: eventuate-local-java-micronaut-jdbc/src/main/java/io/eventuate/local/java/micronaut/jdbc/jdbckafkastore/EventuateLocalFactory.java [hunk 1]
```diff
@@ -52,8 +53,11 @@ public AggregateCrud asyncAggregateCrud(io.eventuate.javaclient.commonimpl.sync.
 
   @Singleton
   public EventuateKafkaAggregateSubscriptions aggregateEvents(EventuateKafkaConfigurationProperties eventuateLocalAggregateStoreConfiguration,
-                                                              EventuateKafkaConsumerConfigurationProperties eventuateKafkaConsumerConfigurationProperties) {
-    return new EventuateKafkaAggregateSubscriptions(eventuateLocalAggregateStoreConfiguration, eventuateKafkaConsumerConfigurationProperties);
+                                                              EventuateKafkaConsumerConfigurationProperties eventuateKafkaConsumerConfigurationProperties,
+                                                              KafkaConsumerFactory kafkaConsumerFactory) {
+    return new EventuateKafkaAggregateSubscriptions(eventuateLocalAggregateStoreConfiguration,
+            eventuateKafkaConsumerConfigurationProperties,
+            kafkaConsumerFactory);
```
Comment (cer) [inline eventuate-local-java-micronaut-jdbc/src/main/java/io/eventuate/local/java/micronaut/jdbc/jdbckafkastore/EventuateLocalFactory.java:60]: What's the purpose of this change? Seems unrelated to classloader issue.

Comment (dartartem) [reply eventuate-local-java-micronaut-jdbc/src/main/java/io/eventuate/local/java/micronaut/jdbc/jdbckafkastore/EventuateLocalFactory.java:60]: Chris, now kafka messaging project contains this change: https://github.com/eventuate-foundation/eventuate-messaging-kafka/pull/35/files
It makes necessary to have KafkaConsumerFactory to created consumers in eventuate local and tram.
And when I tested changes I also updated eventuate-messaging-kafka project with snapshot of eventuate-common
to avoid any conflicts between versions of eventuate-common in eventuate local, eventuate tram, eventuate tram sagas.
Anyway we need to update all libraries step by step and this change is required.
Please let me remind you how dependencies look like:
eventuate-common->eventuate-common-messaging->eventuate-messaging-kafka

eventuate-common->eventuate-local
eventuate-messaging-kafka->eventuate-local

eventuate-common->eventuate-tram-core
eventuate-messaging-kafka->eventuate-tram-core

eventuate-common->eventuate-tram-sagas
eventuate-messaging-kafka->eventuate-tram-sagas

Comment (cer) [reply eventuate-local-java-micronaut-jdbc/src/main/java/io/eventuate/local/java/micronaut/jdbc/jdbckafkastore/EventuateLocalFactory.java:60]: Ok.  But the PR description just says that it's about class loading. Please update the description.

File: eventuate-client-java-micronaut-jdbc/src/main/java/io/eventuate/javaclient/micronaut/jdbc/EmbeddedTestAggregateStoreFactory.java [hunk 2]
```diff
@@ -14,38 +16,43 @@
...
 import javax.sql.DataSource;
+import java.util.Collections;
 
 @Factory
 public class EmbeddedTestAggregateStoreFactory {
 
-  @Primary
-  public EventuateCommonJdbcOperations eventuateCommonJdbcOperations(EventuateJdbcStatementExecutor eventuateJdbcStatementExecutor) {
-    return new EventuateCommonJdbcOperations(eventuateJdbcStatementExecutor);
+  @Singleton
```
Comment (cer) [inline eventuate-client-java-micronaut-jdbc/src/main/java/io/eventuate/javaclient/micronaut/jdbc/EmbeddedTestAggregateStoreFactory.java:29]: This looks suspicious.
I don't think a library should define `PlatformTransactionManager` - Perhaps the tests should?

File: eventuate-client-java-spring-jdbc/src/main/java/io/eventuate/javaclient/spring/jdbc/EmbeddedTestAggregateStoreConfiguration.java [hunk 3]
```diff
@@ -37,8 +38,21 @@ public EventuateDatabaseScriptSupplier eventuateCommonInMemoryScriptSupplierForE
   }
 
   @Bean
-  public EventuateJdbcAccess eventuateJdbcAccess(EventuateTransactionTemplate eventuateTransactionTemplate, EventuateJdbcStatementExecutor eventuateJdbcStatementExecutor, EventuateCommonJdbcOperations eventuateCommonJdbcOperations) {
-    return new EventuateJdbcAccessImpl(eventuateTransactionTemplate, eventuateJdbcStatementExecutor, eventuateCommonJdbcOperations);
+  public TransactionTemplate transactionTemplate(DataSource dataSource) {
```
Comment (cer) [inline eventuate-client-java-spring-jdbc/src/main/java/io/eventuate/javaclient/spring/jdbc/EmbeddedTestAggregateStoreConfiguration.java:41]: This looks suspicious too for the same reasons as mentioned earlier.
